src/sql/util.py

```diff
@@ -157,6 +157,84 @@ def show_deprecation_warning():
     )
 
 
+def check_duplicate_arguments(
+    magic_execute, cmd_from, args, allowed_duplicates
+) -> bool:
+    """
+    Raises UsageError when duplicate arguments are passed to magics.
+    Returns true if no duplicates in arguments or aliases.
+    """
+
+    aliased_arguments = {}
+    unaliased_arguments = []
+
+    for decorator in magic_execute.decorators:
+        assert isinstance(decorator.args, tuple)
+        dec_args = decorator.args
+        if len(dec_args) > 1:
+            aliased_arguments[dec_args[0]] = dec_args[1]
+        else:
+            if dec_args[0].startswith("--") or dec_args[0].startswith("-"):
+                unaliased_arguments.append(dec_args[0])
+
+    if len(args) <= 1:
+        return True
+
+    args = [arg for arg in args if arg.startswith("--") or arg.startswith("-")]
+
+    single_hyphen_opts = set()
+    double_hyphen_opts = set()
+
+    for arg in args:
+        if arg.startswith("--"):
+            double_hyphen_opts.add(arg)
+        elif arg.startswith("-"):
+            single_hyphen_opts.add(arg)
+
+    duplicate_args = []
```

Comment:
Please add some comments here also

Reply:
Improved readability in [this](https://github.com/ploomber/jupysql/pull/908/commits/b5db9f347004e94f68d64f3e76a1244fd895c0a8) commit